mirrord/operator/src/setup.rs

```diff
@@ -0,0 +1,276 @@
+use std::{collections::BTreeMap, convert::Infallible, io::Write, str::FromStr};
+
+use k8s_openapi::{
+    api::{
+        apps::v1::{Deployment, DeploymentSpec},
+        core::v1::{
+            Container, ContainerPort, EnvVar, Namespace, PodSpec, PodTemplateSpec, ServiceAccount,
+        },
+        rbac::v1::{ClusterRole, ClusterRoleBinding, PolicyRule, RoleRef, Subject},
+    },
+    apimachinery::pkg::apis::meta::v1::{LabelSelector, ObjectMeta},
+};
+use thiserror::Error;
+
+/// General Operator Error
+#[derive(Debug, Error)]
+pub enum SetupError {
+    #[error(transparent)]
+    Reader(#[from] std::io::Error),
+    #[error(transparent)]
+    YamlSerialization(#[from] serde_yaml::Error),
+}
+
+type Result<T, E = SetupError> = std::result::Result<T, E>;
+
+pub trait OperatorSetup {
```

Comment:
Shouldn't this be called like Serializable?

Reply:
A bit too generic of a name Serializable, I wanted it to be more specific to the setup